src/hotspot/share/gc/parallel/psPromotionManager.inline.hpp

```diff
@@ -247,9 +247,21 @@ inline oop PSPromotionManager::copy_unmarked_to_survivor_space(oop o,
   // Copy obj
   Copy::aligned_disjoint_words(cast_from_oop<HeapWord*>(o), cast_from_oop<HeapWord*>(new_obj), new_obj_size);
 
-  // Parallel GC claims with a release - so other threads might access this object
-  // after claiming and they should see the "completed" object.
-  ContinuationGCSupport::transform_stack_chunk(new_obj);
+  if (UseCompactObjectHeaders) {
+    // The copy above is not atomic. Make sure we have seen the proper mark
+    // and re-install it into the copy, so that Klass* is guaranteed to be correct.
+    markWord mark = o->mark_acquire();
```

Comment:
I don't think we need the acquire here, do we?

Reply:
Right. An atomic load would be sufficient, which is what oopDesc::mark() already does. I change those code paths to use just mark().